File: solutions/day-06/day06Jokot.kt
class Solution {
    fun productExceptSelf(nums: IntArray): IntArray {
        val size = nums.size
        val output = IntArray(size) { 1 }

        for (i in 1 until size) {
            output[i] = output[i-1] * nums[i-1]
        }

        var suffix = 1
        for (i in size - 1 downTo 0) {
            output[i] *= suffix
            suffix *= nums[i]
        }

        return output
    }
}

File: solutions/day-06/day06Jokot.py
class Solution:
    def productExceptSelf(self, nums: list[int]) -> list[int]:
        size = len(nums)
        output = [1] * size

        for i in range(1, size):
            output[i] = nums[i-1] * output[i-1]
        
        temp_suffix = 1
        for i in range(size-1, -1, -1):
            output[i] *= temp_suffix
            temp_suffix *= nums[i]

        return output